Still failing.

```
Traceback (most recent call last):
  File "./wpt", line 5, in <module>
    wpt.main()
  File "/home/travis/build/w3c/web-platform-tests/tools/wpt/wpt.py", line 129, in main
    rv = script(*args, **kwargs)
  File "/home/travis/build/w3c/web-platform-tests/tools/ci/check_stability.py", line 355, in run
    iterations, results, inconsistent = run(venv, wpt_logger, **wpt_kwargs)
  File "/home/travis/build/w3c/web-platform-tests/tools/wpt/stability.py", line 188, in run
    wptrunner.run_tests(**kwargs)
  File "/home/travis/build/w3c/web-platform-tests/tools/wptrunner/wptrunner/wptrunner.py", line 260, in run_tests
    logger.suite_end()
  File "/home/travis/build/w3c/web-platform-tests/tools/wptrunner/wptrunner/environment.py", line 120, in __exit__
    cm.__exit__()
TypeError: __exit__() takes exactly 4 arguments (1 given)
```

--- a/tools/ci/ci_stability.sh
+++ b/tools/ci/ci_stability.sh
@@ -8,7 +8,7 @@ cd $WPT_ROOT
 source tools/ci/lib.sh
 
 test_stability() {
-    ./wpt check-stability $PRODUCT --output-bytes $((1024 * 1024 * 3)) --metadata ~/meta/
+    ./wpt check-stability $PRODUCT --output-bytes $((1024 * 1024 * 3)) --metadata ~/meta/ --install-fonts
 }
 
 main() {

--- a/tools/wptrunner/wptrunner/browsers/sauce.py
+++ b/tools/wptrunner/wptrunner/browsers/sauce.py
@@ -161,7 +161,7 @@ class SauceConnect():
         if self.sc_process.returncode is not None and self.sc_process.returncode > 0:
             raise SauceException("Unable to start Sauce Connect Proxy. Process exited with code %s", self.sc_process.returncode)
 
-    def __exit__(self, *args):
+    def __exit__(self, exc_type, exc_val, exc_tb):
         self.sc_process.terminate()
         if os.path.exists(self.temp_dir):
             try:

--- a/tools/wptrunner/wptrunner/environment.py
+++ b/tools/wptrunner/wptrunner/environment.py
@@ -117,7 +117,7 @@ class TestEnvironment(object):
             for port, server in servers:
                 server.kill()
         for cm in self.env_extras:
-            cm.__exit__()
+            cm.__exit__(exc_type, exc_val, exc_tb)
         self.cache_manager.__exit__(exc_type, exc_val, exc_tb)
         self.ssl_env.__exit__(exc_type, exc_val, exc_tb)
         self.stash.__exit__()

--- a/tools/wptrunner/wptrunner/font.py
+++ b/tools/wptrunner/wptrunner/font.py
@@ -1,0 +1,110 @@
+import ctypes
+import logging
+import os
+import platform
+
+from shutil import copy2, rmtree
+from subprocess import call
+
+HERE = os.path.split(__file__)[0]
+SYSTEM = platform.system().lower()
+
+
+class FontInstaller(object):
+    def __init__(self, font_dir=None, **fonts):
+        self.font_dir = font_dir
+        self.installed_fonts = False
+        self.created_dir = False
+        self.fonts = fonts
+
+    def __enter__(self, options=None):
+        for _, font_path in self.fonts.items():
+            font_name = font_path.split('/')[-1]
+            install = getattr(self, 'install_%s_font' % SYSTEM, None)
+            if not install:
+                logging.warning('Font installation not supported on %s',
+                                SYSTEM)
+                return False
+            if install(font_name, font_path):
+                self.installed_fonts = True
+                logging.info('Installed font: %s', font_name)
+            else:
+                logging.warning('Unable to install font: %s', font_name)
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if not self.installed_fonts:
+            return False
+
+        for _, font_path in self.fonts.items():
+            font_name = font_path.split('/')[-1]
+            remove = getattr(self, 'remove_%s_font' % SYSTEM, None)
+            if not remove:
+                logging.warning('Font removal not supported on %s', SYSTEM)
+                return False
+            if remove(font_name, font_path):
+                logging.info('Removed font: %s', font_name)
+            else:
+                logging.warning('Unable to remove font: %s', font_name)
+
+    def install_linux_font(self, font_name, font_path):
+        if not self.font_dir:
+            self.font_dir = os.path.join(os.path.expanduser('~'), '.fonts')
+        if not os.path.exists(self.font_dir):
+            os.makedirs(self.font_dir)
+            self.created_dir = True
+        if not os.path.exists(os.path.join(self.font_dir, font_name)):
+            copy2(font_path, self.font_dir)
+        try:
+            fc_cache_returncode = call('fc-cache')
+            return not fc_cache_returncode
+        except OSError:  # If fontconfig doesn't exist, return False
+            logging.error('fontconfig not available on this Linux system.')
+            return False
+
+    def install_darwin_font(self, font_name, font_path):
+        if not self.font_dir:
+            self.font_dir = os.path.join(os.path.expanduser('~'),
+                                         'Library/Fonts')
+        if not os.path.exists(self.font_dir):
+            os.makedirs(self.font_dir)
+            self.created_dir = True
+        if not os.path.exists(os.path.join(self.font_dir, font_name)):
+            copy2(font_path, self.font_dir)
+        return True
+
+    def install_windows_font(self, _, font_path):
+        hwnd_broadcast = 0xFFFF
+        wm_fontchange = 0x001D
+
+        gdi32 = ctypes.WinDLL('gdi32')
+        if gdi32.AddFontResourceW(font_path):
+            return bool(ctypes.windll.user32.SendMessageW(hwnd_broadcast,
+                                                          wm_fontchange))
+
+    def remove_linux_font(self, font_name, _):
+        if self.created_dir:
+            rmtree(self.font_dir)
+        else:
+            os.remove('%s/%s' % (self.font_dir, font_name))
+        try:
+            fc_cache_returncode = call('fc-cache')
+            return not fc_cache_returncode
+        except OSError:  # If fontconfig doesn't exist, return False
+            logging.error('fontconfig not available on this Linux system.')
+            return False
+
+    def remove_darwin_font(self, font_name, _):
+        if self.created_dir:
+            rmtree(self.font_dir)
+        else:
+            os.remove(os.path.join(self.font_dir, font_name))
+        return True
+
+    def remove_windows_font(self, _, font_path):
+        hwnd_broadcast = 0xFFFF
+        wm_fontchange = 0x001D
+
+        gdi32 = ctypes.WinDLL('gdi32')
+        if gdi32.RemoveFontResourceW(font_path):
+            return bool(ctypes.windll.user32.SendMessageW(hwnd_broadcast,
+                                                          wm_fontchange))

--- a/tools/wptrunner/wptrunner/wptcommandline.py
+++ b/tools/wptrunner/wptrunner/wptcommandline.py
@@ -138,6 +138,11 @@ scheme host and port.""")
                               default=None, help="Browser against which to run tests")
     config_group.add_argument("--config", action="store", type=abs_path, dest="config",
                               help="Path to config file")
+    config_group.add_argument("--install-fonts", action="store_true",
+                              default=None,
+                              help="Allow the wptrunner to install fonts on your system")
+    config_group.add_argument("--font-dir", action="store", type=abs_path, dest="font_dir",
+                              help="Path to local font installation directory", default=None)
 
     build_type = parser.add_mutually_exclusive_group()
     build_type.add_argument("--debug-build", dest="debug", action="store_true",

--- a/tools/wptrunner/wptrunner/wptrunner.py
+++ b/tools/wptrunner/wptrunner/wptrunner.py
@@ -10,6 +10,7 @@ import testloader
 import wptcommandline
 import wptlogging
 import wpttest
+from font import FontInstaller
 from testrunner import ManagerGroup
 from browsers.base import NullBrowser
 
@@ -141,6 +142,12 @@ def run_tests(config, test_paths, product, **kwargs):
 
         check_args(**kwargs)
 
+        if kwargs["install_fonts"]:
+            env_extras.append(FontInstaller(
+                font_dir=kwargs["font_dir"],
+                ahem=os.path.join(kwargs["tests_root"], "fonts/Ahem.ttf")
+            ))
+
         if "test_loader" in kwargs:
             run_info = wpttest.get_run_info(kwargs["run_info"], product, debug=None,
                                             extras=run_info_extras(**kwargs))
@@ -251,7 +258,6 @@ def run_tests(config, test_paths, product, **kwargs):
                 if repeat_until_unexpected and unexpected_total > 0:
                     break
                 logger.suite_end()
-
     return unexpected_total == 0
 
 def start(**kwargs):

